libkbfs/folder_branch_ops.go

```diff
@@ -1176,6 +1176,16 @@ func (fbo *folderBranchOps) SetInitialHeadFromServer(
 		return fmt.Errorf("MD with revision=%d not initialized", md.Revision())
 	}
 
+	// Return early if the head is already set.  This avoids taking
+	// mdWriterLock for no reason.
+	lState := makeFBOLockState()
+	head := fbo.getHead(lState)
+	if head != (ImmutableRootMetadata{}) && head.mdID == md.mdID {
```

Comment:
I think you don't even need the `mdID` check, since this function doesn't do anything if `head` is set, period.